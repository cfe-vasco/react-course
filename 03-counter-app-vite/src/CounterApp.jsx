import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {

  const [counter, setCounter] = useState(value);

  const add = () => {
    setCounter(counter + 1); // ===> Toma el valor del counter y lo aumenta en 1
    // setCounter ( (c) => c + 1 ) ===> esto es una funcion que regresa el nuevo valor del counter
  };

  const remove = () => setCounter(counter - 1)

  const reset = () => setCounter (value)

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={add}>+1</button>
      <button onClick={remove}>-1</button>
      <button aria-label='btn-reset' onClick={reset}>Reset</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
