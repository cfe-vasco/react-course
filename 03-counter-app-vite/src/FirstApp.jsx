// const user = {
//   name: 'Christopher Vasco',
//   age: 22,
// }

const getPass = (edad) => {
  return (edad >= 18) ? 'Puedes entrar a la discoteca' : 'Eres menor de edad'
}

export const FirstApp = () => {
  return (
    <>
      <h1>Bienvenido</h1>
      <p>{getPass(19)}</p>
      {/* <code>JSON.stringify{user}</code> */}
    </>
  );
};