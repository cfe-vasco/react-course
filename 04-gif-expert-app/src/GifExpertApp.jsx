import { useState } from 'react';
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch', 'Dragon Ball']);

  const onAddCategory = () => {

    // const newCategory = 'Valorant'
    // if(categories.includes(newCategory)){
    //     console.log('Esta categoria ya existe');
    // } else {
    //     setCategories([...categories, newCategory])
    // }

    // setCategories(newCategory => [...newCategory, 'Valorant'])

    setCategories(['Valorant', ...categories])

  }

  return (
    <>
      {/* titulo */}
      <h1>GifExpertApp</h1>

      {/* Input */}
      <AddCategory setCategories={setCategories}/>
      {/* Listado de Gifs */}
      <ol>
        {categories.map((category) => {
          return <li key={category}>{category}</li>;
        })}
      </ol>
      {/* Gif Item */}
    </>
  );
};
