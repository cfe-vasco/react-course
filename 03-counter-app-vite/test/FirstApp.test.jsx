import { render } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en FirstApp', () => {
    test('Debe de hacer match con snapshot', () => {

        const subTitle = 'Esto es un subtitulo desde tests'
        const {container} = render(<FirstApp subTitle={subTitle}/>)

        expect(container).toMatchSnapshot()

    })

    test('Debe de mostrar el titulo en h1', () => {
        const subTitle = 'Esto es un subtitulo desde tests 2'
        const {container, getByText} = render(<FirstApp subTitle={subTitle}/>)
        expect(getByText(subTitle) ).toBeTruthy()

        const h1 = container.querySelector('p')
        expect(h1.innerHTML).toContain(subTitle)
    })
})