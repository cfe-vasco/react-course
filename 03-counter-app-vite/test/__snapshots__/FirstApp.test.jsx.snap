// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe de hacer match con snapshot 1`] = `
<div>
  <h1>
    Hola 
    Christopher Vasco
  </h1>
  <h2>
    No hay titulo
  </h2>
  <p>
    Esto es un subtitulo desde tests
  </p>
</div>
`;
